### DISCUSSION TITLE
Disaster Recovery Guidance

### DISCUSSION THREAD

**Discussion by dtdionne** [solidusio/solidus#5706] [Troubleshooting]
I haven’t found any docs or guidance on backup and recovery.  We’ve done some testing and it hasn’t gone well.

We switched to Postgres and basically made a store with products and stuff.  We GitHub the application and backup the db.  Then simulate a bare metal restore and it’s quite wonky.  After cloning and restoring the db some categories and taxons make it but no products, breadcrumbs seem tainted with shrapnel from the sample store and the store name reverts to “sample store” even though it’s correct in settings.

Any docs or guidance on DR?

Thanks

**Comment by jarednorman** [comment]
Hey, I converted this to a Discussion since it wasn't strictly speaking an "issue" with Solidus itself. I didn't see a really good category so I just threw it in Troubleshooting. 🤷🏻 

At the heart of it, Solidus apps are Rails apps. All best practices when it comes to backups, monitoring, and resiliency apply. My best guess is you're experiencing some caching issues, but without knowing more about your caching setup it's hard to say.

Normally Solidus apps have relatively few moving parts that store any kind of state and need to be worried about:

- The Database, which you should continuous protection on if you're using Heroku Postgres
- The caching server (memcached, Redis, etc.), but if you're doing a full restore you should nuke it
- The object storage (S3, etc.), resiliency/restoration is really specific to which service you use, whether you're using ActiveStorage, Paperclip, Shrine, or something else
- Job Queues (Redis for Sidekiq, other things for other backends)
